src/posix/platform/openthread-posix-config.h

```diff
@@ -407,4 +407,13 @@
 #define OPENTHREAD_POSIX_CONFIG_RCP_CAPS_DIAG_ENABLE OPENTHREAD_CONFIG_DIAG_ENABLE
 #endif
 
+/**
+ * @def OPENTHREAD_POSIX_CONFIG_UPSTREAM_DNS_BIND_TO_INFRA_NETIF
+ *
+ * Define as 1 to let the upstream DNS bind the socket to infra network interface.
+ */
+#ifndef OPENTHREAD_POSIX_CONFIG_UPSTREAM_DNS_BIND_TO_INFRA_NETIF
+#define OPENTHREAD_POSIX_CONFIG_UPSTREAM_DNS_BIND_TO_INFRA_NETIF 0
```

Comment:
What is the purpose of this build config? Should we always bind to the infrastructure interface?

Reply:
On CPE devices, the infrastructure interface is the downlink (LAN), but we should actually send DNS requests to uplink (WAN), in this case, follow the system routing table would be a better solution.

Reply:
I suspect most TBRs will be Wi-Fi hosts. Should we default to assume using the infrastructure interface?

Reply:
updated the default to enabled